sycl/include/sycl/ext/oneapi/experimental/sycl_complex.hpp

```diff
@@ -498,9 +498,10 @@ template <class _Tp> struct __libcpp_complex_overload_traits<_Tp, false, true> {
 
 // real
 
-template <class _Tp, class = std::enable_if<is_gencomplex<_Tp>::value>>
-__DPCPP_SYCL_EXTERNAL _SYCL_EXT_CPLX_INLINE_VISIBILITY constexpr _Tp
-real(const complex<_Tp> &__c) {
+template <class _Tp>
+__DPCPP_SYCL_EXTERNAL _SYCL_EXT_CPLX_INLINE_VISIBILITY constexpr
+    typename std::enable_if_t<is_genfloat<_Tp>::value, _Tp>
```

Comment:
Is `typename` needed here?

Reply:
Not really. Without `typename` it was still compiling due to the deduction guides in C++17 which allow it to work even without `typename`.
However, using `typename` ensures portability and compatibility with older C++ versions, as well as readability and coherence in this header (since most `enable::if` were using `typename`)

Reply:
I don't think we support anything less than C++17...

Reply:
Right, I guess my argument now, is just coherence with the codebase and `cppreference`.

Would you prefer me to remove all references of `typename` ?

Reply:
Gentle ping in case you did not see my response.

@steffenlarsen @aelovikov-intel :point_up:

Reply:
@steffenlarsen is on vacation for another day or two. I'm going to merge as-is and we can resolve this discussion post-commit.